coprocessor/fhevm-engine/fhevm-engine-common/src/healthz_server.rs

```diff
@@ -0,0 +1,135 @@
+use axum::{
+    extract::State,
+    http::StatusCode,
+    response::{IntoResponse, Json},
+    routing::{get, Router},
+};
+use serde::Serialize;
+use sqlx::PgPool;
+use std::{collections::HashMap, net::SocketAddr, sync::Arc};
+use tokio::net::TcpListener;
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info};
+
+#[derive(Serialize)]
+struct HealthResponse {
+    status_code: String,
+    status: String,
+    fields: HashMap<String, bool>,
```

Comment:
why bool ?

Reply:
in that case maybe rename fields to checks

Reply:
Done.
However, these checks are serialized as `dependencies` in the HTTP response.